tests/deploy/e2e/test_code_loading.py

```diff
@@ -855,7 +886,123 @@ async def check_code_for_version(
     await check_code_for_version(
         version=5,
         environment=environment,
+        codemanager=codemanager,
         agent_names=["agent_Z", "agent_A"],
         module_name="new_module",
         expected_source=b"#Yet some other code",
     )
+
+
+@pytest.mark.parametrize("auto_start_agent", [True])
+async def test_project_constraints_in_agent_code_install(server, client, environment, clienthelper):
```

Comment:
* This test doesn't test whether the constraint is actually taken into account, right?
* I think we must also add a test to verify whether the exporter works correctly.
* We probably also want to explicitly test the behavior when a partial export is done.